### PR TITLE
fix june trigger

### PR DESCRIPTION
**Pull request by suhailkakar**
**What does this pull request do? Explain your changes. (required)**
Preivously june events used to trigger multiple times, this PR adds callback to make sure each event is being sent just once

**Specific updates (required)**

- Updated the implementation and used usedCallback hook

**How did you test each of these updates (required)**

- Manually though browser network tab and June dashboard

**Does this pull request close any open issues?**

**Screenshots (optional)**

**Checklist**

- [ ] I have read the **CONTRIBUTING** document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: packages/www/components/Sidebar/index.tsx [hunk 1]
```diff
@@ -73,9 +74,16 @@ export type SidebarId =
...
+
   const June = useJune();
 
-  June?.track(`sidebar ${id}`);
+  const trackSidebarEvent = useCallback(() => {
+    if (June) June.track(`sidebar ${id}`);
+  }, [June]);
+
+  useEffect(() => {
+    trackSidebarEvent();
```
Comment (0xcadams) [inline packages/www/components/Sidebar/index.tsx]: Could you explain what this useEffect is supposed to do? Is it on mount we fire this event?

**Review by 0xcadams — APPROVED** [review]
LGTM just one question on the useEffect

File: packages/www/components/Sidebar/index.tsx (hunk 1 above)
Comment (suhailkakar) [reply packages/www/components/Sidebar/index.tsx]: Yeah! just want to make sure the event is fired once the component is mounted. Do you think we can update with a better approach? I don't like using useEffect here as well :)

Comment (0xcadams) [reply packages/www/components/Sidebar/index.tsx]: What's being tracked here? A user is shown a sidebar option?

Comment (suhailkakar) [reply packages/www/components/Sidebar/index.tsx]: Yes, the sidebar id in which the page is active, like `stream`, `asset`, `developers` is being tracked and sent to June

Comment (0xcadams) [reply packages/www/components/Sidebar/index.tsx]: Okay maybe we use pageview events for this?
https://www.june.so/docs/quickstart/install

Comment (suhailkakar) [reply packages/www/components/Sidebar/index.tsx]: yeah, i've checked that, but they don't have that method on their SDK. We might need to add their API request then. 
Should I do it @0xcadams?

Comment (0xcadams) [reply packages/www/components/Sidebar/index.tsx]: Oh odd - maybe we just add a route change handler instead with the events (supported by their SDK)?

```
const router = useRouter(); 

  useEffect(() => {
    const handleRouteChange = (url, { shallow }) => {
      if (analytics) analytics.page(url);
    };

    router.events.on('routeChangeStart', handleRouteChange);

    return () => {
      router.events.off('routeChangeStart', handleRouteChange);
    };
  }, [analytics]);
  ```

Comment (suhailkakar) [reply packages/www/components/Sidebar/index.tsx]: yep! sounds good! just updated it.

Comment (0xcadams) [reply packages/www/components/Sidebar/index.tsx]: LGTM!

**Pull request merged**